script/axe-docs

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+# This workflow is intended to help catch accesibility violations by running axe on every documentation example.
+# Developers frequently copy and paste examples from documentation so it's important to ensure the examples are
+# accessible.
+# Learn more about @axe-core/cli here: https://github.com/dequelabs/axe-core-npm/tree/develop/packages/cli
+
+npm install -g @axe-core/cli
+
+# https://stackoverflow.com/a/21189044
+parse_yaml() {
+   local prefix=$2
+   local s='[[:space:]]*' w='[a-zA-Z0-9_]*' fs=$(echo @|tr @ '\034')
+   sed -ne "s|^\($s\)\($w\)$s:$s\"\(.*\)\"$s\$|\1$fs\2$fs\3|p" \
+        -e "s|^\($s\)\($w\)$s:$s\(.*\)$s\$|\1$fs\2$fs\3|p"  $1 |
+   awk -F$fs '{
+      indent = length($1)/2;
+      vname[indent] = $2;
+      for (i in vname) {if (i > indent) {delete vname[i]}}
+      if (length($3) > 0) {
+         vn=""; for (i=0; i<indent; i++) {vn=(vn)(vname[i])("_")}
+         printf("%s%s%s=\"%s\"\n", "'$prefix'",vn, $2, $3);
+      }
+   }'
+}
+
+# https://stackoverflow.com/a/8574392
+containsElement () {
+  local e match="$1"
+  shift
+  for e; do [[ "$e" == "$match" ]] && return 0; done
+  return 1
+}
+
+# Finds all paths for component or utilities pages because they contain examples.
+nav_paths=($(parse_yaml docs/src/@primer/gatsby-theme-doctocat/nav.yml | grep -e "/components/" -e "/utilities/" | awk -F "=" '{print $2}'))
+
+# Starting point violations
+# DO NOT ADD TO THIS LIST!
+needs_to_be_fixed=(
+  /components/autocomplete
+  /components/avatars
+  /components/box
+  /components/buttons
+  /components/header
+  /components/markdown
+  /components/progress
+  /components/select-menu
+  /components/labels
+  /components/timeline
+  /components/toasts
+  /utilities/flexbox
+  /utilities/layout 
+)
+
+echo "############################################################"
+echo "WARNING"
+echo "The following paths will be excluded from this automated accessibility check."
+echo "They contain known violations and should be addressed:"
+printf '%s\n' "${needs_to_be_fixed[@]}"
+echo "############################################################"
+
+# Generate doc urls 
+doc_urls=()
+for i in "${nav_paths[@]}"; 
+do
+  if containsElement "${i//\"/}" "${needs_to_be_fixed[@]}"; then
+    continue
+  else
+    doc_url="primer.style/css$i"
+    doc_url="${doc_url//\"/}"
+    doc_urls+=($doc_url)
+  fi
+done
+
+# https://github.com/dequelabs/axe-core-npm/tree/develop/packages/cli
+# We exclude rules that depend on full page context.
```

Comment:
We also disable `color-contrast` because there's too many that addressing it will be a blocker

Reply:
I think that is a common practice.